zerver/actions/message_send.py

```diff
@@ -817,14 +817,6 @@ def do_send_messages(
 
             ums.extend(user_messages)
 
-            send_request.message.service_queue_events = get_service_bot_events(
-                sender=send_request.message.sender,
-                service_bot_tuples=send_request.service_bot_tuples,
-                mentioned_user_ids=mentioned_user_ids,
-                active_user_ids=send_request.active_user_ids,
-                recipient_type=send_request.message.recipient.type,
-            )
```

Comment:
I think the right fix is to just have this be an attribute on `send_request`, not no `message` -- I think we just missed this one when we moved all the rendering time metadata to this separate `send_request` object.

Then there's no need to adjust the transaction structure.

Reply:
I see. I thought of creating a `dataclass` subclassing the `SendMessageRequest` that has one additional `service_queue_events` attribute because conceptually the caller does not need to fill this out. Though adding the attribute directly should work as well.